src/App.tsx

```diff
@@ -1,10 +1,120 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import { FC, useCallback, useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos, uploadTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+
+enum Errors {
+  LOAD_TODOS = 'Unable to load todos',
+  VALIDATION = 'Title should not be empty',
+  ADD = 'Unable to add a todo',
+  DELETE = 'Unable to delete a todo',
+  UPDATE = 'Unable to update a todo',
+}
+
+enum TodoStatus {
+  All = 'All',
+  Active = 'Active',
+  Completed = 'Completed',
+}
+
+const TodoStatusRoutes: Record<TodoStatus, string> = {
+  [TodoStatus.All]: '/',
+  [TodoStatus.Active]: '/active',
+  [TodoStatus.Completed]: '/completed',
+};
+
+const emptyTodo: Omit<Todo, 'id'> = {
+  completed: false,
+  userId: USER_ID,
+  title: '',
+};
+
+const errorDelay = 3000;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTodo, setNewTodo] = useState<Omit<Todo, 'id'>>(emptyTodo);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>('');
+  const [selectedStatus, setSelectedStatus] = useState<TodoStatus>(
+    TodoStatus.All,
+  );
+
+  const closeErrorHandler = () => {
+    setError('');
+  };
+
+  const filteringTodosByStatus = useMemo(() => {
+    switch (selectedStatus) {
+      case TodoStatus.Active:
+        return [...todos].filter(todo => todo.completed === false);
```

Comment:
```suggestion
        return todos.filter(todo => todo.completed === false);
```
filter doesn't mutate the array